qcp/grovers_algorithm.py

```diff
@@ -0,0 +1,125 @@
+from qcp.matrices import DefaultMatrix
+import random
+import qcp.constants as c
+import qcp.tensor_product as tp
```

Comment:
Once you remove these to lines to appease the pep8 gods, I'll be happy to approve the PR